tests/filter/test_subscribe_update.py

```diff
@@ -0,0 +1,107 @@
+            try:
+                self.check_published_message_reaches_filter_peer(message)
+            except Exception as ex:
+                logger.error(f'ContentTopic {content_topic["description"]} failed: {str(ex)}')
+                failed_content_topics.append(content_topic)
+        assert not failed_content_topics, f"ContentTopics failed: {failed_content_topics}"
+
+    def test_filter_update_subscription_add_31_new_content_topics(self, subscribe_main_nodes):
+        try:
+            _31_content_topics = [input["value"] for input in SAMPLE_INPUTS[:31]]
+            self.update_filter_subscription({"requestId": "1", "contentFilters": _31_content_topics, "pubsubTopic": self.test_pubsub_topic})
+            raise AssertionError("Subscribe with more than 30 content topics worked!!!")
+        except Exception as ex:
+            assert "Bad Request" in str(ex)
+
+    def test_filter_update_subscription_refresh_existing(self):
+        self.wait_for_subscriptions_on_main_nodes([self.test_content_topic], pubsub_topic=self.test_pubsub_topic)
+        self.update_filter_subscription({"requestId": "1", "contentFilters": [self.test_content_topic], "pubsubTopic": self.test_pubsub_topic})
+        self.check_published_message_reaches_filter_peer(self.create_message(contentTopic=self.test_content_topic))
+
+    def test_filter_update_subscription_add_a_new_pubsub_topic(self):
+        self.wait_for_subscriptions_on_main_nodes([self.test_content_topic], pubsub_topic=self.test_pubsub_topic)
+        self.update_filter_subscription(
+            {"requestId": "1", "contentFilters": [self.test_content_topic, self.second_conted_topic], "pubsubTopic": VALID_PUBSUB_TOPICS[4]}
+        )
+        self.add_new_relay_subscription(VALID_PUBSUB_TOPICS[4:5])
+        self.check_published_message_reaches_filter_peer(
+            self.create_message(contentTopic=self.test_content_topic), pubsub_topic=self.test_pubsub_topic
+        )
+        self.check_published_message_reaches_filter_peer(
+            self.create_message(contentTopic=self.test_content_topic), pubsub_topic=VALID_PUBSUB_TOPICS[4]
+        )
+        self.check_published_message_reaches_filter_peer(
+            self.create_message(contentTopic=self.second_conted_topic), pubsub_topic=VALID_PUBSUB_TOPICS[4]
+        )
+
+    def test_filter_update_subscription_with_no_pubsub_topic(self, subscribe_main_nodes):
+        try:
+            self.update_filter_subscription({"requestId": "1", "contentFilters": [self.test_content_topic]})
+            raise AssertionError(f"Subscribe with no pubusub topics worked!!!")
+        except Exception as ex:
+            assert "Bad Request" in str(ex)
+
+    def test_filter_update_subscription_with_invalid_pubsub_topic_format(self, subscribe_main_nodes):
+        try:
+            self.update_filter_subscription({"requestId": "1", "contentFilters": [self.test_content_topic], "pubsubTopic": [self.test_pubsub_topic]})
```

Comment:
With "invalid pubsub topic form" we're talking about "type" or "string formatting"? 
I mean: "list of strings" (which is what seems to happen) vs the "specified rfc string format"?

I don't know if I'm making sense 😅

Reply:
Yes, in this test I'm passing a list of strings instead of string. I will update the test name to be more suggestive